scene/resources/material.cpp

```diff
@@ -684,29 +684,35 @@ void BaseMaterial3D::_update_shader() {
 		return; //no update required in the end
 	}
 
-	MutexLock lock(shader_map_mutex);
-	if (shader_map.has(current_key)) {
-		shader_map[current_key].users--;
-		if (shader_map[current_key].users == 0) {
-			// Deallocate shader which is no longer in use.
-			RS::get_singleton()->free(shader_map[current_key].shader);
-			shader_map.erase(current_key);
+	{
+		MutexLock lock(shader_map_mutex);
+		if (ShaderData *v = shader_map.getptr(current_key); v) {
```

Comment:
```suggestion
		ShaderData *v = shader_map.getptr(current_key);
		if (v) {
```
We like the old fashioned style more

Reply:
I didn't get any feedback, so I've switched to the preferred style.